components/ntp_sponsored_images/browser/BUILD.gn

```diff
@@ -0,0 +1,39 @@
+source_set("browser") {
+  assert_no_deps = [ "//chrome/*" ]
+
+  sources = [
+    "locale_helper.cc",
+    "locale_helper.h",
+    "locale_helper_linux.cc",
+    "locale_helper_linux.h",
+    "locale_helper_mac.mm",
+    "locale_helper_mac.h",
+    "locale_helper_win.cc",
+    "locale_helper_win.h",
+    "ntp_sponsored_images_component_manager.cc",
+    "ntp_sponsored_images_component_manager.h",
+    "ntp_sponsored_images_data.cc",
+    "ntp_sponsored_images_data.h",
+    "ntp_sponsored_images_internal_data.cc",
+    "ntp_sponsored_images_internal_data.h",
+    "ntp_sponsored_images_service.cc",
+    "ntp_sponsored_images_service.h",
+    "ntp_sponsored_image_source.cc",
+    "ntp_sponsored_image_source.h",
+    "switches.cc",
+    "switches.h",
+    "regional_component_data.cc",
+    "regional_component_data.h",
+    "url_constants.cc",
+    "url_constants.h",
+  ]
+
+  deps = [
+    "//base",
+    "//brave/components/brave_component_updater/browser",
+    "//components/component_updater",
+    "//components/keyed_service/core",
```

Comment:
is this really the full list? Aren't there references to prefs, etc..?

Reply:
Yes, it's full list. `ntp_sponsored_images` component only refers these dependencies.

Reply:
@bridiver The classes which have more deps are in `brave/browser/ntp_sponsored_images`